common/src/main/java/com/microsoft/identity/common/nativeauth/internal/controllers/NativeAuthMsalController.kt

```diff
@@ -1217,6 +1388,25 @@ class NativeAuthMsalController : BaseNativeAuthController() {
         )
     }
 
+    private fun performSubmitChallengeTokenRequest(
+        oAuth2Strategy: NativeAuthOAuth2Strategy,
+        parameters: MFASubmitChallengeCommandParameters
+    ): SignInTokenApiResult {
+        LogSession.logMethodCall(
+            tag = TAG,
+            correlationId = parameters.getCorrelationId(),
+            methodName = "${TAG}.performOOBTokenRequest"
+        )
+        // Transform MFASubmitChallengeCommandParameters into SignInSubmitCodeCommandParameters,
```

Comment:
@nilo-ms this is why I don't like submitChallenge(): the API call is the same for submit code. So in order to support challenges that are not `oob`, we'll need to make changes to the SDK anyway.

Reply:
I see. We will do something similar for iOS too. Although I'm not particularly fan of the term "challenge" either, we opted for it as a more general alternative to "code". If we feel strongly about it, we could reconsider and discuss this with Kaushik.